packages/eslint-plugin/tests/rules/await-thenable.test.ts

```diff
@@ -475,5 +518,99 @@ for  (const value of yieldNumberPromises()) {
         },
       ],
     },
+    {
+      code: `
+declare const disposable: Disposable;
+async function foo() {
+  await using d = disposable;
+}
+      `,
+      errors: [
+        {
+          column: 19,
+          endColumn: 29,
+          endLine: 4,
+          line: 4,
+          messageId: 'awaitUsingOfNonAsyncDisposable',
+          suggestions: [
+            {
+              messageId: 'removeAwait',
+              output: `
+declare const disposable: Disposable;
+async function foo() {
+   using d = disposable;
+}
+      `,
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: `
+async function foo() {
+  await using _ = {
+    async [Symbol.dispose]() {},
+  };
+}
+      `,
+      errors: [
+        {
+          column: 19,
+          endColumn: 4,
+          endLine: 5,
+          line: 3,
+          messageId: 'awaitUsingOfNonAsyncDisposable',
+          suggestions: [
+            {
+              messageId: 'removeAwait',
+              output: `
+async function foo() {
+   using _ = {
+    async [Symbol.dispose]() {},
+  };
+}
+      `,
+            },
+          ],
+        },
+      ],
+    },
+    {
+      code: `
```

Comment:
Honestly I'm kind of torn on this one. In principle, the user could split it up to be 
```ts
  using a = disposable,
  await using b = asyncDisposable,
  using c = disposable,
  await using d = asyncDisposable,
  using e = disposable;
```

But it doesn't really have any benefit (unlike how the `for await` case can be actively harmful to use `for await` unnecessarily). So we could change the logic to make this a valid case unless _none_ of the declarators are possibly-async-iterable 🤷

Reply:
No strong preference from me. I think it's fine as-is 🤷.